src/OpenFga.Sdk/Client/Client.cs

```diff
@@ -0,0 +1,409 @@
+        ClientReadAuthorizationModelOptions? options = default,
+        CancellationToken cancellationToken = default) {
+        var authorizationModelId = GetAuthorizationModelId(options);
+        if (authorizationModelId == null) {
+            throw new FgaRequiredParamError("ClientConfiguration", "AuthorizationModelId");
+        }
+
+        return await api.ReadAuthorizationModel(authorizationModelId, cancellationToken);
+    }
+
+    /**
+     * ReadLatestAuthorizationModel - Read the latest authorization model for the current store
+     */
+    public async Task<ReadAuthorizationModelResponse> ReadLatestAuthorizationModel(
+        IClientRequestOptionsWithAuthZModelId? options = default,
+        CancellationToken cancellationToken = default) {
+        var response =
+            await ReadAuthorizationModels(new ClientReadAuthorizationModelsOptions { PageSize = 1 }, cancellationToken);
+        return new ReadAuthorizationModelResponse { AuthorizationModel = response.AuthorizationModels?[0] };
+    }
+
+    /***********************
+     * Relationship Tuples *
+     ***********************/
+
+    /**
+     * Read Changes - Read the list of historical relationship tuple writes and deletes
+     */
+    public async Task<ReadChangesResponse> ReadChanges(ClientReadChangesRequest body,
+        ClientReadChangesOptions? options = default,
+        CancellationToken cancellationToken = default) =>
+        await api.ReadChanges(body.Type, options?.PageSize, options?.ContinuationToken, cancellationToken);
+
+    /**
+     * Read - Read tuples previously written to the store (does not evaluate)
+     */
+    public async Task<ReadResponse> Read(ClientReadRequest body, ClientReadOptions? options = default,
+        CancellationToken cancellationToken = default) =>
+        await api.Read(
+            new ReadRequest {
+                TupleKey = body,
+                PageSize = options?.PageSize,
+                ContinuationToken = options?.ContinuationToken
+            }, cancellationToken);
+
+    /**
+     * Write - Create or delete relationship tuples
+     */
+    public async Task<ClientWriteResponse> Write(ClientWriteRequest body, ClientWriteOptions? options = default,
+        CancellationToken cancellationToken = default) {
+        var maxPerChunk =
+            options.Transaction?.MaxPerChunk ??
+            1; // 1 has to be the default otherwise the chunks will be sent in transactions
+        var maxParallelReqs =
+            options.Transaction?.MaxParallelRequests ?? DEFAULT_MAX_METHOD_PARALLEL_REQS;
+        var authorizationModelId = GetAuthorizationModelId(options);
+
+        if (options.Transaction?.Disable != true) {
+            await api.Write(
+                new WriteRequest {
+                    Writes = new TupleKeys(body.Writes ?? new List<TupleKey>()),
+                    Deletes = new TupleKeys(body.Deletes ?? new List<TupleKey>()),
```

Comment:
if either `writes` or `deletes` is empty, e.g.

```c#
var body = new ClientWriteRequest() {
       Deletes = new List<TupleKey> {
              new("document:roadmap", "viewer", "user:81684243-9356-4421-8fbf-a4f8d36aa31b")
        },
 };
```

 this will send an empty JSON array

```json
{"writes":{"tuple_keys":[]},"deletes":{"tuple_keys":[{"object":"document:roadmap","relation":"viewer","user":"user:81684243-9356-4421-8fbf-a4f8d36aa31b"}]},"authorization_model_id":"01GXC8W24QFB04PR2553HEXP05"}
```


 which then gives

```
invalid TupleKeys.TupleKeys: value must contain at least 1 item(s)
```